### ISSUE TITLE
AQL does not permit COLLECT with AGGREGATE and INTO ... KEEP

### ISSUE THREAD

**Issue by Simran-B** [arangodb/arangodb#3488]
Arango: v3.2.5 stable
OS: Win10 64bit

The following query fails with
> Query: AQL: **syntax error**, unexpected identifier, expecting { near 'r RETURN {color: c, count: c...' at position 8:70 (while parsing)

```js
LET data = [
    {color: "blue", rating: 5, attr: true},
    {color: "red", rating: 2, attr: true},
    {color: "red", rating: 1, attr: true}
]
FOR e IN data
    LET r = e.rating
    COLLECT c = e.color AGGREGATE count = LENGTH(1) INTO groups KEEP r
    RETURN {color: c, count: count, rating: groups[*].r}
```

It works if `KEEP r` is removed, but both variables `e` and `r` and thus a lot more data ends up in `groups`, although it's not needed later in the query.

If AGGREGATE is replaced by post-counting, `INTO ... KEEP` works just fine:

```js
FOR e IN data
    LET r = e.rating
    COLLECT c = e.color INTO groups KEEP r
    RETURN {color: c, count: LENGTH(groups), rating: groups[*].r}
```

A projection `AGGREGATE ... INTO var = value` does also work and eliminates `e` from `groups`:

```js
FOR e IN data
    LET r = e.rating
    COLLECT c = e.color AGGREGATE count = LENGTH(1) INTO groups = r
    RETURN {color: c, count: count, rating: groups}
```

Hence, the only combination which is not allowed is `AGGREGATE ... INTO ... KEEP`. Is this intentional?

**Comment by jsteemann** [comment]
@Simran-B : yes, it is intentional. Not because it makes any sense syntax-wise, but only because it simplified the already very complex parser grammar for AQL COLLECT.

**Comment by Simran-B** [comment]
Do you think that it would make sense to support it from a technical point of view?

We should definitely improve the documentation to make it easier to understand for users how COLLECT works, and I also talked with @MrPieces about a possible data aggregation course.
